packages/@tinacms/cli/src/cmds/compile/index.ts

```diff
@@ -81,32 +80,46 @@ export const compile = async (_ctx, _next) => {
 
 const transpile = async (projectDir, tempDir) => {
   logger.info(logText('Transpiling...'))
-  // Make sure that post paths are posix (unix paths). This is necessary on windows.
-  const posixProjectDir = normalize(projectDir)
-  const posixTempDir = normalize(tempDir)
 
-  return Promise.all(
-    glob
-      // We will replaces \\ with / as required by docs see: https://github.com/mrmlnc/fast-glob#how-to-write-patterns-on-windows
-      .sync(path.join(projectDir, '**', '*.ts').replace(/\\/g, '/'), {
-        ignore: [
-          path
-            .join(projectDir, '__generated__', '**', '*.ts')
-            .replace(/\\/g, '/'),
-        ],
-      })
-      .map(async function (file) {
-        const fullPath = path.resolve(file)
-
-        const contents = await fs.readFileSync(fullPath).toString()
-        const newContent = ts.transpile(contents)
-        const newPath = file
-          .replace(posixProjectDir, posixTempDir)
-          .replace('.ts', '.js')
-        await fs.outputFile(newPath, newContent)
-        return true
-      })
+  const packageJSON = JSON.parse(
+    fs.readFileSync(packageJSONFilePath).toString() || '{}'
   )
+  const deps = packageJSON?.dependencies || []
```

Comment:
Is this the correct way to go about the external modules @jeffsee55 ?

Reply:
It is, but I don't think want to exclude them, it works fine for now - but if/when the `defineSchema` moves of to `tinacms` we actually need to "bundle" this so our bundled output includes everything we need (specifically the `defineSchema` function). See comment for `esbuild-plugin-pnp`. At the very least we'll want to be sure to include `tinacms`, if it's problematic to do more than that then we can cross that bridge when we get there.

Reply:
Makes sense